src/pages/products/Products.js

```diff
@@ -32,23 +40,45 @@ class Products extends Component {
     });
   }
 
+  deleteProduct = (id) => {
+    this.props.dispatch(deleteProduct(id));
+  }
+
+  onClickHand = () => {
+    const { history } = this.props;
+
+    event.preventDefault();
+    history.push('products/new');
+  }
+
   render() {
     const { listProd } = this.state;
```

Comment:
why do you need it in state? Products come from properties, why don't use it directly?